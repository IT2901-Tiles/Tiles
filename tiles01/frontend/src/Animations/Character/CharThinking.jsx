import React from "react"
import Character from './Character'

function CharThinking() {
    return (
        <div className='CharThinking'>
            <svg viewBox="0 0 165 150" className="svgCharacter">
                <Character/> 
                <g className="LightBulb">
                    <g className="LightBulbAnimation">
                        <line id="lightStick1" x1="62" y1="14" x2="55" y2="10" stroke="black" stroke-width="2"/>
                        <line id="lightStick2" x1="65" y1="9" x2="60" y2="3" stroke="yellow" stroke-width="2"/>
                        <line id="lightStick3" x1="69" y1="7" x2="68" y2="1" stroke="black" stroke-width="2"/>
                        <line id="lightStick4" x1="73" y1="8" x2="76" y2="3" stroke="yellow" stroke-width="2"/>
                        <circle id="lightBulb" cx="70" cy="14" r="5" fill="yellow" stroke="black"/>                    
                        <line id="lightStickMain2" x1="84" y1="20.5" x2="75" y2="15" stroke="black" stroke-width="2"/>
                        <animateTransform
                            attributeName = "transform"
                            attributeType = "XML"
                            type = "rotate"
                            keyTimes = "0; 0.25; 0.5; 0.75; 1"
                            values = "-2 84 20.5;-15 84 20.5; -2 84 20.5; -15 84 20.5; -2 84 20.5;"
                            begin = "0s"
                            dur = "4s"
                            repeatCount = "indefinite"
                        />
                    </g>
                <line id="lightStickMain1" x1="87.5" y1="28" x2="84" y2="20" stroke="black" stroke-width="2"/>
                </g>
               
                    <circle id="leftEye" cx="75" cy="50" r="7" stroke="black">
                        <animate
                            attributeName = "r"
                            keyTimes="0;0.5;1"
                            values="7;5;7"
                            dur="2s"                        
                            repeatCount="indefinite"
                        />
                    </circle>
                    <circle id="rightEye" cx="92" cy="50" r="5" stroke="black">
                        <animate
                            attributeName = "r"
                            keyTimes="0;0.5;1"
                            values="5;7;5"
                            dur="2s"                        
                            repeatCount="indefinite"
                        />
                    </circle>

                <g className="leftArm"> 
                    <circle id="shoulderL" cx="53.5" cy="86" r="5" fill="grey" stroke="black" />
                    <rect id="overarmL" x="55" y="67" height="16" width="7" fill="grey" stroke="black" transform="rotate(-120 55 89.5)"/>
                    <circle id="elbowL" cx="28" cy="101" r="5" fill="grey" stroke="black" />
                    <rect id="underarmL" x="24" y="78" height="16" width="6.5" fill="grey" stroke="black" transform="rotate(18 28 101)"/>
                    <rect id="handLL" x="30" y="62" height="14" width="5" fill="grey" stroke="black" transform="rotate(-10 32.5 68.5)"/>
                    <rect id="handLR" x="39" y="62" height="14" width="5" fill="grey" stroke="black" transform="rotate(30 39.5 68.5)"/>
                    <animateTransform
                        attributeName = "transform"
                        attributeType = "XML"
                        type = "rotate"
                        keyTimes = "0; 0.25; 0.5; 0.60; 0.80; 1"
                        values = "5 53.5 86;40 53.5 86; 40 53.5 86; 40 53.5 86; 5 53.5 86;5 53.5 86;"
                        begin = "0s"
                        dur = "5s"
                        repeatCount = "indefinite"
                    />
                </g>
                <g className="rightArm">
                    <circle id="shoulderR" cx="121.5" cy="86" r="5" fill="grey" stroke="black" />           
                    <rect id="overarmR" x="118" y="93" height="16" width="7" fill="grey" stroke="black"/>
                    <circle id="elbowR" cx="121.5" cy="115.5" r="5" fill="grey" stroke="black" />
                    <rect id="underarmR" x="128" y="113" height="7" width="16" fill="grey" stroke="black" transform="rotate(-20 121.5 115.5)"/>
                    <rect id="handRL" x="140" y="103" height="5" width="13" fill="grey" stroke="black" transform="rotate(-30 140 101)"/>
                    <rect id="handRR" x="145" y="107" height="5" width="13" fill="grey" stroke="black" transform="rotate(10 145 107)"/> 
                </g> 
                <g className="thinkingCloud">
                    <circle cx="10" cy="14" r="8" fill="white"/>
                    <circle cx="20" cy="11" r="8" fill="white"/>
                    <circle cx="30" cy="14" r="8" fill="white"/>
                    <circle cx="40" cy="16" r="6" fill="white"/>
                    <circle cx="4" cy="18" r="3.5" fill="white"/>
                    <circle cx="12" cy="20" r="8" fill="white"/>
                    <circle cx="22" cy="22" r="8" fill="white"/>
                    <circle cx="32" cy="26" r="6" fill="white"/>
                    <circle cx="38" cy="22" r="4" fill="white"/>

                    <g className="thinkingBubbles">
                        <circle cx="35" cy="38" r="3" fill="white">
                            <animate
                                attributeName = "fill"
                                keyTimes="0;0.5;0.8;1"
                                values="white;#68C2C4;white;white"
                                dur="3s"                        
                                repeatCount="indefinite"
                            /> 
                        </circle>
                        <circle cx="42" cy="44" r="2" fill="white"> 
                            <animate
                                attributeName = "fill"
                                keyTimes="0;0.3;1"
                                values="#68C2C4;white;white"
                                dur="3s"                        
                                repeatCount="indefinite"
                            />
                        </circle>
                    </g>

                </g> 
            </svg>
        </div>
    )
}

export default CharThinking;
