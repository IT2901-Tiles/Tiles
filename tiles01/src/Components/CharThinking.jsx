import React from "react"
import Character from './Character'

function CharThinking() {
    return (
        <div className='CharThinking'>

            <svg height="150" width="165">
                <Character/> 
                <g className="LightBulb">
                <line id="lightStick1" x1="62" y1="14" x2="55" y2="10" stroke="black" stroke-width="2"/>
                <line id="lightStick2" x1="65" y1="9" x2="60" y2="3" stroke="yellow" stroke-width="2"/>
                <line id="lightStick3" x1="69" y1="7" x2="68" y2="1" stroke="black" stroke-width="2"/>
                <line id="lightStick4" x1="73" y1="8" x2="76" y2="3" stroke="yellow" stroke-width="2"/>
                <circle id="lightBulb" cx="70" cy="14" r="5" fill="yellow" stroke="black"/>                    <line id="lightStickMain1" x1="87.5" y1="28" x2="84" y2="20" stroke="black" stroke-width="2"/>
                <line id="lightStickMain2" x1="84" y1="20.5" x2="75" y2="15" stroke="black" stroke-width="2"/>


                </g>
               
                    <circle id="leftEye" cx="75" cy="50" r="7" stroke="black"/>
                    <circle id="rightEye" cx="92" cy="50" r="5" stroke="black"/>

                <g className="leftArm"> 
                    <circle id="shoulderL" cx="53.5" cy="86" r="5" fill="grey" stroke="black" />
                    <rect id="overarmL" x="55" y="67" height="16" width="7" fill="grey" stroke="black" transform="rotate(-120 55 89.5)"/>
                    <circle id="elbowL" cx="28" cy="101" r="5" fill="grey" stroke="black" />
                    <rect id="underarmL" x="24" y="78" height="16" width="6.5" fill="grey" stroke="black" transform="rotate(18 28 101)"/>
                    <rect id="handLL" x="30" y="62" height="14" width="5" fill="grey" stroke="black" transform="rotate(-10 32.5 68.5)"/>
                    <rect id="handLR" x="39" y="62" height="14" width="5" fill="grey" stroke="black" transform="rotate(30 39.5 68.5)"/>
                </g>  
            </svg>
        </div>
    )
}

export default CharThinking;
