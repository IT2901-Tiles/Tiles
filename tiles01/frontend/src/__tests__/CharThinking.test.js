import {configure, shallow} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';
import {cleanup} from '@testing-library/react';
import CharThinking from '../Components/CharThinking';
import renderer from 'react-test-renderer';


afterEach(() => {
    cleanup();
});

configure({
    adapter: new Adapter()
});

describe('CharThinking', () => {
    const wrapper = shallow(<CharThinking/>)
    it('renders correctly', () => {
        const tree = renderer.create(<CharThinking/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders Character', () => {
        expect(wrapper.find('Character')).toHaveLength(1);
    })
});