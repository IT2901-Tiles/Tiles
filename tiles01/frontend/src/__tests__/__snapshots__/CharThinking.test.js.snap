// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharThinking renders correctly 1`] = `
<div
  className="CharThinking"
>
  <svg
    className="svgCharacter"
    viewBox="0 0 165 150"
  >
    <g
      className="whole"
    >
      <g
        className="HeadParts"
      >
        <rect
          fill="grey"
          height="35"
          id="head"
          stroke="black"
          width="55"
          x="60"
          y="36"
        />
        <rect
          fill="grey"
          height="15"
          id="lEar"
          stroke="black"
          width="5"
          x="53"
          y="46"
        />
        <rect
          fill="grey"
          height="15"
          id="rEar"
          stroke="black"
          width="5"
          x="117"
          y="46"
        />
        <rect
          fill="grey"
          height="6"
          id="topHead"
          stroke="black"
          width="15"
          x="80"
          y="28"
        />
        <circle
          cx="87.5"
          cy="61"
          fill="red"
          id="smile"
          r="8"
          stroke="black"
        />
        <rect
          fill="grey"
          height="10"
          id="smileRect"
          width="19"
          x="78"
          y="52"
        />
        <line
          id="topSmile"
          stroke="black"
          x1="79"
          x2="96"
          y1="62"
          y2="62"
        />
      </g>
      <g
        className="BodyParts"
      >
        <rect
          fill="grey"
          height="8"
          id="neck"
          stroke="black"
          width="15"
          x="80"
          y="73"
        />
        <rect
          fill="grey"
          height="35"
          id="body"
          stroke="black"
          width="55"
          x="60"
          y="83"
        />
      </g>
    </g>
    <g
      className="LightBulb"
    >
      <g
        className="LightBulbAnimation"
      >
        <line
          id="lightStick1"
          stroke="black"
          stroke-width="2"
          x1="62"
          x2="55"
          y1="14"
          y2="10"
        />
        <line
          id="lightStick2"
          stroke="yellow"
          stroke-width="2"
          x1="65"
          x2="60"
          y1="9"
          y2="3"
        />
        <line
          id="lightStick3"
          stroke="black"
          stroke-width="2"
          x1="69"
          x2="68"
          y1="7"
          y2="1"
        />
        <line
          id="lightStick4"
          stroke="yellow"
          stroke-width="2"
          x1="73"
          x2="76"
          y1="8"
          y2="3"
        />
        <circle
          cx="70"
          cy="14"
          fill="yellow"
          id="lightBulb"
          r="5"
          stroke="black"
        />
        <line
          id="lightStickMain2"
          stroke="black"
          stroke-width="2"
          x1="84"
          x2="75"
          y1="20.5"
          y2="15"
        />
        <animateTransform
          attributeName="transform"
          attributeType="XML"
          begin="0s"
          dur="4s"
          keyTimes="0; 0.25; 0.5; 0.75; 1"
          repeatCount="indefinite"
          type="rotate"
          values="-2 84 20.5;-15 84 20.5; -2 84 20.5; -15 84 20.5; -2 84 20.5;"
        />
      </g>
      <line
        id="lightStickMain1"
        stroke="black"
        stroke-width="2"
        x1="87.5"
        x2="84"
        y1="28"
        y2="20"
      />
    </g>
    <circle
      cx="75"
      cy="50"
      id="leftEye"
      r="7"
      stroke="black"
    >
      <animate
        attributeName="r"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="7;5;7"
      />
    </circle>
    <circle
      cx="92"
      cy="50"
      id="rightEye"
      r="5"
      stroke="black"
    >
      <animate
        attributeName="r"
        dur="2s"
        keyTimes="0;0.5;1"
        repeatCount="indefinite"
        values="5;7;5"
      />
    </circle>
    <g
      className="leftArm"
    >
      <circle
        cx="53.5"
        cy="86"
        fill="grey"
        id="shoulderL"
        r="5"
        stroke="black"
      />
      <rect
        fill="grey"
        height="16"
        id="overarmL"
        stroke="black"
        transform="rotate(-120 55 89.5)"
        width="7"
        x="55"
        y="67"
      />
      <circle
        cx="28"
        cy="101"
        fill="grey"
        id="elbowL"
        r="5"
        stroke="black"
      />
      <rect
        fill="grey"
        height="16"
        id="underarmL"
        stroke="black"
        transform="rotate(18 28 101)"
        width="6.5"
        x="24"
        y="78"
      />
      <rect
        fill="grey"
        height="14"
        id="handLL"
        stroke="black"
        transform="rotate(-10 32.5 68.5)"
        width="5"
        x="30"
        y="62"
      />
      <rect
        fill="grey"
        height="14"
        id="handLR"
        stroke="black"
        transform="rotate(30 39.5 68.5)"
        width="5"
        x="39"
        y="62"
      />
      <animateTransform
        attributeName="transform"
        attributeType="XML"
        begin="0s"
        dur="5s"
        keyTimes="0; 0.25; 0.5; 0.60; 0.80; 1"
        repeatCount="indefinite"
        type="rotate"
        values="5 53.5 86;40 53.5 86; 40 53.5 86; 40 53.5 86; 5 53.5 86;5 53.5 86;"
      />
    </g>
    <g
      className="rightArm"
    >
      <circle
        cx="121.5"
        cy="86"
        fill="grey"
        id="shoulderR"
        r="5"
        stroke="black"
      />
      <rect
        fill="grey"
        height="16"
        id="overarmR"
        stroke="black"
        width="7"
        x="118"
        y="93"
      />
      <circle
        cx="121.5"
        cy="115.5"
        fill="grey"
        id="elbowR"
        r="5"
        stroke="black"
      />
      <rect
        fill="grey"
        height="7"
        id="underarmR"
        stroke="black"
        transform="rotate(-20 121.5 115.5)"
        width="16"
        x="128"
        y="113"
      />
      <rect
        fill="grey"
        height="5"
        id="handRL"
        stroke="black"
        transform="rotate(-30 140 101)"
        width="13"
        x="140"
        y="103"
      />
      <rect
        fill="grey"
        height="5"
        id="handRR"
        stroke="black"
        transform="rotate(10 145 107)"
        width="13"
        x="145"
        y="107"
      />
    </g>
    <g
      className="thinkingCloud"
    >
      <circle
        cx="10"
        cy="14"
        fill="white"
        r="8"
      />
      <circle
        cx="20"
        cy="11"
        fill="white"
        r="8"
      />
      <circle
        cx="30"
        cy="14"
        fill="white"
        r="8"
      />
      <circle
        cx="40"
        cy="16"
        fill="white"
        r="6"
      />
      <circle
        cx="4"
        cy="18"
        fill="white"
        r="3.5"
      />
      <circle
        cx="12"
        cy="20"
        fill="white"
        r="8"
      />
      <circle
        cx="22"
        cy="22"
        fill="white"
        r="8"
      />
      <circle
        cx="32"
        cy="26"
        fill="white"
        r="6"
      />
      <circle
        cx="38"
        cy="22"
        fill="white"
        r="4"
      />
      <g
        className="thinkingBubbles"
      >
        <circle
          cx="35"
          cy="38"
          fill="white"
          r="3"
        >
          <animate
            attributeName="fill"
            dur="3s"
            keyTimes="0;0.5;0.8;1"
            repeatCount="indefinite"
            values="white;#68C2C4;white;white"
          />
        </circle>
        <circle
          cx="42"
          cy="44"
          fill="white"
          r="2"
        >
          <animate
            attributeName="fill"
            dur="3s"
            keyTimes="0;0.3;1"
            repeatCount="indefinite"
            values="#68C2C4;white;white"
          />
        </circle>
      </g>
    </g>
  </svg>
</div>
`;
